static/app/views/automations/components/dataConditionNodes.tsx

```diff
@@ -0,0 +1,149 @@
+import {createContext, useContext} from 'react';
+import styled from '@emotion/styled';
+
+import NumberField from 'sentry/components/forms/fields/numberField';
+import SelectField from 'sentry/components/forms/fields/selectField';
+import {t, tct} from 'sentry/locale';
+import {
+  type DataCondition,
+  DataConditionType,
+} from 'sentry/types/workflowEngine/dataConditions';
+
+interface DataConditionNodeProps {
+  condition: Omit<DataCondition, 'condition_group' | 'type' | 'id'>;
+  condition_id: string;
+  onUpdate: (condition: Record<string, any>) => void;
+}
+
+export const DataConditionNodeContext = createContext<DataConditionNodeProps | null>(
+  null
+);
+
+export function useDataConditionNodeContext(): DataConditionNodeProps {
+  const context = useContext(DataConditionNodeContext);
+  if (!context) {
+    throw new Error(
+      'useDataConditionNodeContext was called outside of DataConditionNode'
+    );
+  }
+  return context;
+}
+
+type Node = {
+  label: string;
+  configNode?: () => React.ReactNode;
+};
+
+// TODO: split nodes into separate files
+export const dataConditionNodesMap: Partial<Record<DataConditionType, Node>> = {
+  [DataConditionType.FIRST_SEEN_EVENT]: {
+    label: t('A new issue is created'),
+  },
+  [DataConditionType.REGRESSION_EVENT]: {
+    label: t('A resolved issue becomes unresolved'),
+  },
+  [DataConditionType.REAPPEARED_EVENT]: {
+    label: t('An issue escalates'),
+  },
+  [DataConditionType.AGE_COMPARISON]: {
+    label: t('Compare the age of an issue'),
+    configNode: function ConfigNode() {
+      const {condition, condition_id, onUpdate} = useDataConditionNodeContext();
+      return tct('The issue is [comparison_type] [value] [time]', {
+        comparison_type: (
+          <InlineSelectControl
```

Comment:
Same for extracting these, even if it's just to a local function inside of this component. The return statement would then be much more readable! Something like `tct(text, { comparison_type: <Component1 />, value: <Component2 />, time <Component3 /> })`

Reply:
Extracted them in https://github.com/getsentry/sentry/pull/89233/commits/3d1933f98380f8ca3367263cd30fe7586169a924 !